TheraLang.BLL/Services/NewsService.cs

```diff
@@ -0,0 +1,152 @@
+using TheraLang.DAL.UnitOfWork;
+using TheraLang.DAL.Entities;
+using System.Linq;
+using AutoMapper;
+using Microsoft.AspNetCore.Http;
+using Microsoft.EntityFrameworkCore;
+using TheraLang.BLL.DataTransferObjects.NewsDtos;
+
+namespace TheraLang.BLL.Services
+{
+    public class NewsService : INewsService
+    {
+        private readonly IUnitOfWork _unitOfWork;
+        private readonly IFileService _fileService;
+
+        public NewsService(IUnitOfWork unitOfWork, IFileService fileService)
+        {
+            _unitOfWork = unitOfWork;
+            _fileService = fileService;
+        }
+
+        public async Task<IEnumerable<NewsPreviewDto>> GetAllNews()
+        {
+            var news = await _unitOfWork.Repository<News>().GetAll()
+                .Include(e => e.Author).Include(e => e.UploadedContentImages).ToListAsync();
+
+            var mapper = new MapperConfiguration(cfg =>
+            {
+                cfg.CreateMap<News, NewsPreviewDto>()
+                    .ForMember(m => m.AuthorName, opt => opt.MapFrom(sm => sm.Author.UserName));
+            }
+            ).CreateMapper();
+
+            var newsDtos = mapper.Map<IEnumerable<News>, IEnumerable<NewsPreviewDto>>(news);
+
+            return newsDtos;
+        }
+
+        public async Task<NewsDetailsDto> GetNewsById(int id)
+        {
+            var news = await _unitOfWork.Repository<News>().GetAll().Include(e => e.Author)
+                    .Include(e => e.UploadedContentImages).SingleOrDefaultAsync(n => n.Id == id);
+
+            var mapper = new MapperConfiguration(cfg =>
+            {
+                cfg.CreateMap<News, NewsDetailsDto>()
+                    .ForMember(m => m.ContentImageUrls, opt => opt.MapFrom(sm => sm.UploadedContentImages.Select(i => i.Url)))
+                    .ForMember(m => m.AuthorName, opt => opt.MapFrom(sm => sm.Author.UserName));
+            }).CreateMapper();
+
+            var newsDto = mapper.Map<News, NewsDetailsDto>(news);
+
+            return newsDto;
+        }
+
+        public async Task AddNews(NewsCreateDto newsDto)
+        {
+            var mapper = new MapperConfiguration(cfg => cfg.CreateMap<NewsCreateDto, News>()).CreateMapper();
+
+            var news = mapper.Map<NewsCreateDto, News>(newsDto);
+
+            news.MainImageUrl = (await UploadImage(newsDto.MainImage)).Url;
+            news.UploadedContentImages = new List<UploadedNewsContentImage>();
+            foreach (var image in newsDto.ContentImages)
+            {
+                news.UploadedContentImages.Add(await UploadImage(image));
+            }
+
+            _unitOfWork.Repository<News>().Add(news);
+            await _unitOfWork.SaveChangesAsync();
+        }
+
+        public async Task RemoveNews(int id)
+        {
+            var news = await _unitOfWork.Repository<News>().Get(i => i.Id == id);
+
+            if (news == default)
+            {
+                throw new ArgumentException($"News with id {id} not found!");
```

Comment:
```suggestion
                throw new ArgumentNullException($"News with id {id} not found!");
```

Reply:
Argument of this function is id and if I throw ArgumentNullException it will means that id is null, but is not null, it's just doesn't exist a news record with such id

Reply:
Argument of this function is id, but you are trying to get an object with it, and if there are no items with that ID it will be null.
actulally, this action **await _unitOfWork.Repository<News>().Get(i => i.Id == id);** will return you an object or a defualt value which is NULL.
As soon as you are using VAR and not declaring type itself it will be harder to understand what default value is, so i suggest to do explicit null check instead of **news == default**